### PR TITLE
Implement `sysconf(SC_SYMLOOP_MAX)`

### PR DESCRIPTION
**Pull request by MartinBriza**
This PR does three things:
 - raises kernel symlink loop limit to 8 (because, as BertalanD pointed out, POSIX specifies it should be at least 8)
 - moves the limit constant to the header file
 - accesses the constant to implement the $subj sysconf call

I'll need this constant in #11275 so I figured (after burning a bit of your time with my naive changes) it'll be best to merge this one separately so i get my feet wet with the process here instead of straight up jumping in with that big chonker of a PR.

### CODE REVIEW

File: Kernel/FileSystem/VirtualFileSystem.h [hunk 1]
```diff
@@ -35,6 +35,8 @@ struct UidAndGid {
 class VirtualFileSystem {
     AK_MAKE_ETERNAL
 public:
+    static constexpr int symlink_recursion_limit = 8; // FIXME: increase?
```
Comment (timschumi) [inline Kernel/FileSystem/VirtualFileSystem.h:38]: A link to the appropriate POSIX specification would probably be nice. Also, if we now abide by the specification, the FIXME is probably obsolete.

File: Kernel/API/POSIX/unistd.h [hunk 2]
```diff
@@ -38,6 +38,7 @@ enum {
     _SC_PAGESIZE,
     _SC_GETPW_R_SIZE_MAX,
     _SC_CLK_TCK,
+    _SC_SYMLOOP_MAX /* The number of symbolic links that can be traversed in the resolution of a pathname in the absence of a loop. */
```
Comment (timschumi) [inline Kernel/API/POSIX/unistd.h:41]: We don't have any comments for the other entries. Again, maybe a link to the relevant specification would be more suitable.

**Review by timschumi — COMMENTED** [review]
In addition to the inline feedback, the first has a typo in "BeralanD". Also, please avoid mentioning people in commit messages, as it generates a notification every time that someone uploads that commit to their fork. :^)

**Comment by MartinBriza** [comment]
I removed the `@` from the mention and fixed the spelling. Sorry!

As for the linking to the docs. Do you adding a comment with a link like https://pubs.opengroup.org/onlinepubs/000095399/basedefs/limits.h.html or something else please? Just point me in the direction where to search otherwise please.

**Comment by timschumi** [comment]
> As for the linking to the docs. Do you adding a comment with a link like https://pubs.opengroup.org/onlinepubs/000095399/basedefs/limits.h.html or something else please? Just point me in the direction where to search otherwise please.

A comment with a link to opengroup is what we usually do, yes. The `unistd.h` page lacks a bit in detail, so linking to `limits.h` instead is probably a good idea.

**Comment by BertalanD** [comment]
The tricky thing about POSIX spec links is that Google always seems to show an outdated version of the standard. I'm personally not too keen on LibC spec links, but if you decide to add it, the correct link is <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/limits.h.html>.

**Comment by MartinBriza** [comment]
Thank you. To remain consistent with the rest of the project then, I'll remove the `unistd` comment and add the link to the VFS header.

File: Kernel/API/POSIX/unistd.h (hunk 2 above)
Comment (MartinBriza) [reply Kernel/API/POSIX/unistd.h]: Removed this comment altogether

File: Kernel/FileSystem/VirtualFileSystem.h (hunk 1 above)
Comment (MartinBriza) [reply Kernel/FileSystem/VirtualFileSystem.h]: Removed the fixme, added a link to POSIX docs

**Review by BertalanD — APPROVED** [review]

**Review by timschumi — APPROVED** [review]

File: Kernel/API/POSIX/unistd.h [hunk 3]
```diff
@@ -38,6 +38,7 @@ enum {
     _SC_PAGESIZE,
     _SC_GETPW_R_SIZE_MAX,
     _SC_CLK_TCK,
+    _SC_SYMLOOP_MAX
```
Comment (timschumi) [inline Kernel/API/POSIX/unistd.h:41]: This should probably have a trailing `,`, even if it's not required by the syntax.

Comment (MartinBriza) [reply Kernel/API/POSIX/unistd.h]: Added

**Pull request merged**